_posts/2024-03-25-Announcing-Data-Prepper-2.7.0.md

```diff
@@ -0,0 +1,120 @@
+---
+layout: post
+title:  "Announcing Data Prepper 2.7.0"
+authors:
+- dvenable
+- qichen
+date: 2024-03-25 13:30:00 -0600
+categories:
+  - releases
+meta_keywords: GeoIP enrichment, AWS Secrets Manager Support, XML parsing, ION parsing, decompression
+meta_description: Data Prepper 2.7.0 supports enriching events with GeoIP data, has support for using AWS Secrets Manager, and adds many new processors.
+---
+
+Data Prepper 2.7.0 is now available for [download](https://opensearch.org/downloads.html#data-prepper).
+This release supports extracting geographic locations from IP addresses, supports injectable secrets, and adds many new processors.
+
+
+## GeoIP processor
+
+Data Prepper can now enrich events with geographical location data from an IP address using the new `geoip` processor.
+The `geoip` processor uses the MaxMind GeoLite2 databases to provide geographic location from IP addresses.
+
+Many OpenSearch and Data Prepper users want to enrich their data by adding geographical locations to events.
+There are a number of reasons this data can be valuable.
+Some examples are: looking for anomalies in network access; customer analytics; understanding load across geographies; and more.
+An industry solution for determining a geographical location is through the use of IP addresses.
+
+One example scenario is locating users to a web server.
+Data Prepper already supports parsing Apache Common Log Format for Apache HTTP servers in the `grok` processor.
+In the following example, we can now locate the client making requests using the `clientip` property extracted from the `grok` processor.
+
+```
+processor:
+  - grok:
+      match:
+        log: [ "%{COMMONAPACHELOG_DATATYPED}" ]
+
+  - geoip:
+      entries:
+        - source: clientip
+          target: clientlocation
+          include_fields: [latitude, longitude, location, postal_code, country_name, city_name]
+```
+
+When ingesting data using this pipeline, the OpenSearch index will now contain the geolocation fields expressed above, such as the `latitude` and `city_name`.
+Additionally, you can configure template mappings in OpenSearch so that you can display these events in OpenSearch Dashboards using the [Maps feature](https://opensearch.org/docs/latest/dashboards/visualize/maps/).
+
+
+## AWS Secrets Manager support
+
+Now Data Prepper supports [AWS Secrets Manager](https://aws.amazon.com/secrets-manager/) as extension plugin applicable to pipeline plugins (i.e. source, buffer, processor, sink).
+Users are allowed to configure the AWS secrets extension through extensions in data-prepper-config.yaml.
```

Comment:
Should "Secrets" be capitalized? Should the file name be in code font?

Reply:
I changed to "AWS Secrets Manager"